flask_admin/model/widgets.py

```diff
@@ -135,20 +141,21 @@ def get_kwargs(self, subfield, kwargs):
             kwargs['data-type'] = 'number'
             kwargs['data-step'] = 'any'
         elif subfield.type in ['QuerySelectField', 'ModelSelectField']:
+            # QuerySelectField and ModelSelectField are for relations
             kwargs['data-type'] = 'select'
 
-            choices = {}
+            choices = []
             for choice in subfield:
                 try:
-                    choices[str(choice._value())] = str(choice.label.text)
+                    choices.append({'value': str(choice._value()),
```

Comment:
@mrjoes Fixed. Also had to serialize with json.dumps to get rid of the "u" that's prepended to unicode strings in choices.